### ISSUE TITLE
Errors thrown in useEffect cleanup functions result in `Internal React error`

### ISSUE THREAD

**Issue by nathell** [facebook/react-native#32673]
### Description

When an app includes a functional component which calls `React.useEffect` with a cleanup callback, and that callback throws an error when called, the following is produced in the console:

```
ERROR  Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.
```

The error is not detectable otherwise: the error boundary component does not catch it, nor does installing a global handler with `ErrorUtils.setGlobalHandler()`.

This is on React Native 0.66.3 and React 17.0.2. I’ve also checked React 17.0.1 (no difference in behaviour), and RN 0.65.0 (the one that we currently use in production) which also gobbles up the error, but silently, without producing any message in the console.

The app seemingly continues to work after the error occurs.

I don’t think this is an actual React bug (as opposed to React Native), because on web, the error appears in the browser console, as expected.

I checked only the debug build behaviour, not the release one. It may be the case that in release builds the error sometimes does bubble up to the global error handler (if set), because we’re seeing prod exceptions that are likely to be related to this issue. For the record, what causes this for us is [this issue in react-navigation](https://github.com/react-navigation/react-navigation/issues/9803), exacerbated by the fact that [they had moved removing listeners to a `useEffects` cleanup function](https://github.com/react-navigation/react-navigation/blob/5.x/packages/bottom-tabs/src/utils/useIsKeyboardShown.tsx#L19).

### Version

0.66.3

### Output of `react-native info`

```
System:
    OS: macOS 11.4
    CPU: (6) x64 Intel(R) Core(TM) i5-8500B CPU @ 3.00GHz
    Memory: 1.97 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.3.0 - ~/.nvm/versions/node/v15.3.0/bin/node
    Yarn: 1.22.15 - /usr/local/bin/yarn
    npm: 7.0.14 - ~/.nvm/versions/node/v15.3.0/bin/npm
    Watchman: 2021.10.04.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /Users/nathell/.gem/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK: Not Found
  IDEs:
    Android Studio: 3.6 AI-192.7142.36.36.6392135
    Xcode: 12.5/12E262 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.7 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.3 => 0.66.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

1. Create a RN app with `react-native init myapp`.

2. Edit `App.js` to read:

```js
import React from 'react';
import { Text, TouchableOpacity, View, SafeAreaView } from 'react-native';

const Hello = () => {
  React.useEffect(() => {
    console.log("mount");
    return () => {
      console.log("cleanup");
      throw new Error("KABOOM!");
      console.log("cleanup 2");
    };
  }, []);
  return <Text>Hello!</Text>;
}

const App = () => {
  const [shown, show] = React.useState(false);
  const onPress = () => show(!shown);

  return (
    <SafeAreaView>
      { shown ? <Hello /> : null }
      <TouchableOpacity onPress={onPress}>
        <Text>Touch me</Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
};

export default App;
```

3. Run the resulting app (on either iOS or Android) and tap `Touch me` twice.

Expected result (in the `react-native start` output): 
``` 
 LOG  mount
 LOG  cleanup
 ERROR  Kaboom!
```

Actual result:
```
 LOG  mount
 LOG  cleanup
 ERROR  Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.
```

**Comment by nathell** [comment]
Out of curiosity, I checked React 16.13.1. Same behaviour in RN.

I’ve also checked that, in browser, React 17.0.2 (but not 16.13.1) gobbles up the error as well when using a production build:

| | React 16.13.1 | React 17.0.2 |
| --- | --- | --- |
| Development | ✅  Throws error | ✅  Throws error |
| Production | ✅  Throws error | ❌  Captures error |

Experimental code (save as `index.html` and open in browser; replace `production.min.js` with `development.js` and `17` with `16.3` as needed):
```html
<div id="root"></div>
<script src="https://unpkg.com/react@17/umd/react.development.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@17/umd/react-dom.development.js" crossorigin></script>
<script>
const h = React.createElement;

const Hello = () => {
  React.useEffect(() => {
    console.log('mount');
    return () => {
      console.log('cleanup');
      throw new Error('KABOOM!');
      console.log('cleanup2');
    }
  }, []);
  return 'Hello!';
}

const App = () => {
  const [shown, show] = React.useState(true);
  return h('div', null, [
    shown ? h(Hello, {}, null) : null,
    h('button', {onClick: () => show(!shown)}, 'Toggle')
  ]);
}

ReactDOM.render(
  h(App, {}, null),
  document.getElementById("root")
);
</script>
```

Possibly related: https://github.com/facebook/react/issues/22650

**Comment by mehdinourollah** [comment]
Any updates?

**Comment by KrisLau** [comment]
@nathell Did you ever figure out

**Comment by Kailash-MAF** [comment]
Any update regarding fix ?

**Comment by AftabUfaq** [comment]
any solution for this

**Comment by Kailash-MAF** [comment]
Full error

```
Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.

Error message:

TypeError: undefined is not a function
    in BottomTabBar (created by BottomTabView)
    in SafeAreaProviderCompat (created by BottomTabView)
    in BottomTabView (created by BottomTabNavigator)
    in BottomTabNavigator (created by BottomTabs)
    in BottomTabs (created by SceneView)
    in StaticContainer
    in EnsureSingleNavigator (created by SceneView)
    in SceneView (created by RouteView)
    in RCTView (created by View)
    in View (created by DebugContainer)
    in DebugContainer (created by MaybeNestedStack)
    in MaybeNestedStack (created by RouteView)
    in RNSScreen (created by AnimatedComponent)
    in AnimatedComponent
    in AnimatedComponentWrapper (created by InnerScreen)
    in Suspender (created by Freeze)
    in Suspense (created by Freeze)
    in Freeze (created by DelayedFreeze)
    in DelayedFreeze (created by InnerScreen)
    in InnerScreen (created by RouteView)
    in RouteView (created by NativeStackViewInner)
    in Suspender (created by Freeze)
    in Suspense (created by Freeze)
    in Freeze (created by DelayedFreeze)
    in DelayedFreeze (created by ScreenStack)
    in RNSScreenStack (created by ScreenStack)
    in ScreenStack (created by NativeStackViewInner)
    in NativeStackViewInner (created by NativeStackView)
    in RCTView (created by View)
    in View (created by SafeAreaInsetsContext)
    in SafeAreaProviderCompat (created by NativeStackView)
    in NativeStackView (created by NativeStackNavigator)
    in NativeStackNavigator (created by InsideStack)
    in InsideStack (created by SceneView)
    in StaticContainer
    in EnsureSingleNavigator (created by SceneView)
    in SceneView (created by CardContainer)
    in RCTView (created by View)
    in View (created by CardContainer)
    in RCTView (created by View)
    in View (created by CardContainer)
    in RCTView (created by View)
    in View
    in CardSheet (created by Card)
    in RCTView (created by View)
    in View (created by AnimatedComponent)
    in AnimatedComponent
    in AnimatedComponentWrapper (created by PanGestureHandler)
    in PanGestureHandler (created by PanGestureHandler)
    in PanGestureHandler (created by Card)
    in RCTView (created by View)
    in View (created by AnimatedComponent)
    in AnimatedComponent
    in AnimatedComponentWrapper (created by Card)
    in RCTView (created by View)
    in View (created by Card)
    in Card (created by CardContainer)
    in CardContainer (created by CardStack)
    in RNSScreen (created by AnimatedComponent)
    in AnimatedComponent
    in AnimatedComponentWrapper (created by InnerScreen)
    in Suspender (created by Freeze)
    in Suspense (created by Freeze)
    in Freeze (created by DelayedFreeze)
    in DelayedFreeze (created by InnerScreen)
    in InnerScreen (created by Screen)
    in Screen (created by MaybeScreen)
    in MaybeScreen (created by CardStack)
 ERROR  Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.

```

**Comment by AftabUfaq** [comment]
> Full error
> 
> ```
> Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.
> 
> Error message:
> 
> TypeError: undefined is not a function
>     in BottomTabBar (created by BottomTabView)
>     in SafeAreaProviderCompat (created by BottomTabView)
>     in BottomTabView (created by BottomTabNavigator)
>     in BottomTabNavigator (created by BottomTabs)
>     in BottomTabs (created by SceneView)
>     in StaticContainer
>     in EnsureSingleNavigator (created by SceneView)
>     in SceneView (created by RouteView)
>     in RCTView (created by View)
>     in View (created by DebugContainer)
>     in DebugContainer (created by MaybeNestedStack)
>     in MaybeNestedStack (created by RouteView)
>     in RNSScreen (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by InnerScreen)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by InnerScreen)
>     in InnerScreen (created by RouteView)
>     in RouteView (created by NativeStackViewInner)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by ScreenStack)
>     in RNSScreenStack (created by ScreenStack)
>     in ScreenStack (created by NativeStackViewInner)
>     in NativeStackViewInner (created by NativeStackView)
>     in RCTView (created by View)
>     in View (created by SafeAreaInsetsContext)
>     in SafeAreaProviderCompat (created by NativeStackView)
>     in NativeStackView (created by NativeStackNavigator)
>     in NativeStackNavigator (created by InsideStack)
>     in InsideStack (created by SceneView)
>     in StaticContainer
>     in EnsureSingleNavigator (created by SceneView)
>     in SceneView (created by CardContainer)
>     in RCTView (created by View)
>     in View (created by CardContainer)
>     in RCTView (created by View)
>     in View (created by CardContainer)
>     in RCTView (created by View)
>     in View
>     in CardSheet (created by Card)
>     in RCTView (created by View)
>     in View (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by PanGestureHandler)
>     in PanGestureHandler (created by PanGestureHandler)
>     in PanGestureHandler (created by Card)
>     in RCTView (created by View)
>     in View (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by Card)
>     in RCTView (created by View)
>     in View (created by Card)
>     in Card (created by CardContainer)
>     in CardContainer (created by CardStack)
>     in RNSScreen (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by InnerScreen)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by InnerScreen)
>     in InnerScreen (created by Screen)
>     in Screen (created by MaybeScreen)
>     in MaybeScreen (created by CardStack)
>  ERROR  Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.
> ```

this us an issue with native base

**Comment by Kailash-MAF** [comment]
> Full error
> 
> ```
> Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.
> 
> Error message:
> 
> TypeError: undefined is not a function
>     in BottomTabBar (created by BottomTabView)
>     in SafeAreaProviderCompat (created by BottomTabView)
>     in BottomTabView (created by BottomTabNavigator)
>     in BottomTabNavigator (created by BottomTabs)
>     in BottomTabs (created by SceneView)
>     in StaticContainer
>     in EnsureSingleNavigator (created by SceneView)
>     in SceneView (created by RouteView)
>     in RCTView (created by View)
>     in View (created by DebugContainer)
>     in DebugContainer (created by MaybeNestedStack)
>     in MaybeNestedStack (created by RouteView)
>     in RNSScreen (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by InnerScreen)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by InnerScreen)
>     in InnerScreen (created by RouteView)
>     in RouteView (created by NativeStackViewInner)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by ScreenStack)
>     in RNSScreenStack (created by ScreenStack)
>     in ScreenStack (created by NativeStackViewInner)
>     in NativeStackViewInner (created by NativeStackView)
>     in RCTView (created by View)
>     in View (created by SafeAreaInsetsContext)
>     in SafeAreaProviderCompat (created by NativeStackView)
>     in NativeStackView (created by NativeStackNavigator)
>     in NativeStackNavigator (created by InsideStack)
>     in InsideStack (created by SceneView)
>     in StaticContainer
>     in EnsureSingleNavigator (created by SceneView)
>     in SceneView (created by CardContainer)
>     in RCTView (created by View)
>     in View (created by CardContainer)
>     in RCTView (created by View)
>     in View (created by CardContainer)
>     in RCTView (created by View)
>     in View
>     in CardSheet (created by Card)
>     in RCTView (created by View)
>     in View (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by PanGestureHandler)
>     in PanGestureHandler (created by PanGestureHandler)
>     in PanGestureHandler (created by Card)
>     in RCTView (created by View)
>     in View (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by Card)
>     in RCTView (created by View)
>     in View (created by Card)
>     in Card (created by CardContainer)
>     in CardContainer (created by CardStack)
>     in RNSScreen (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by InnerScreen)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by InnerScreen)
>     in InnerScreen (created by Screen)
>     in Screen (created by MaybeScreen)
>     in MaybeScreen (created by CardStack)
>  ERROR  Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.
> ```

@satya164 Do you have any solution regarding this ?

**Comment by AftabUfaq** [comment]
@Kailash-MAF trying to find out this issue. but didnt find out

**Comment by walid-mos** [comment]
I got roughly the same issue here, with react base 

```
Error Stack : 
Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.

Error message:

TypeError: undefined is not a function
    in HybridProvider (created by NativeBaseProvider)
    in ResponsiveQueryProvider (created by NativeBaseProvider)
    in RNCSafeAreaProvider (created by SafeAreaProvider)
    in SafeAreaProvider (created by NativeBaseProvider)
    in NativeBaseConfigProviderProvider (created by NativeBaseProvider)
    in NativeBaseProvider (created by App)
    in Provider (created by App)
    in App
```

**Comment by NaashNix** [comment]
I got the same error! 

`Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.

Error message:

TypeError: undefined is not a function
    in HybridProvider (created by NativeBaseProvider)
    in ResponsiveQueryProvider (created by NativeBaseProvider)
    in RNCSafeAreaProvider (created by SafeAreaProvider)
    in SafeAreaProvider (created by NativeBaseProvider)
    in NativeBaseConfigProviderProvider (created by NativeBaseProvider)
    in NativeBaseProvider (created by Register)
    in Register (created by SceneView)
    in StaticContainer
    in EnsureSingleNavigator (created by SceneView)
    in SceneView (created by SceneView)
    in RCTView (created by View)
    in View (created by DebugContainer)
    in DebugContainer (created by MaybeNestedStack)
    in MaybeNestedStack (created by SceneView)
    in RNSScreen (created by AnimatedComponent)
    in AnimatedComponent
    in AnimatedComponentWrapper (created by InnerScreen)
    in Suspender (created by Freeze)
    in Suspense (created by Freeze)
    in Freeze (created by DelayedFreeze)
    in DelayedFreeze (created by InnerScreen)
    in InnerScreen (created by Screen)
    in Screen (created by SceneView)
    in SceneView (created by NativeStackViewInner)
    in Suspender (created by Freeze)
    in Suspense (created by Freeze)
    in Freeze (created by DelayedFreeze)
    in DelayedFreeze (created by ScreenStack)`

**Comment by yirenkeji555** [comment]
solved it ?

**Comment by mcanikhilprajapati** [comment]
same issue facing here

**Comment by aphillipo** [comment]
I also have this issue...

**Comment by hendradedis** [comment]
yap me too , have this issue , anyone can solve it ?

**Comment by Tran281** [comment]
Check if any async is used inside useEffect hook and remove it.
Worked for me

**Comment by KrisLau** [comment]
The return in a `useEffect` should be a function: https://reactjs.org/docs/hooks-effect.html#:~:text=Effects%20with%20Cleanup. If you're getting an error, check what you're returning in the `useEffect`

**Comment by thong-dipro** [comment]
Same problems.
Device Infor:
```
Window 11
Android 12
```

**Comment by grant-solomons** [comment]
I am experiencing the same problem. The error gets thrown when I fast refresh but not when I enter the page

**Comment by grant-solomons** [comment]
> I am experiencing the same problem. The error gets thrown when I fast refresh but not when I enter the page

I removed async and await from a useEffect

**Comment by Degan90** [comment]
> Full error
> 
> ```
> Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.
> 
> Error message:
> 
> TypeError: undefined is not a function
>     in BottomTabBar (created by BottomTabView)
>     in SafeAreaProviderCompat (created by BottomTabView)
>     in BottomTabView (created by BottomTabNavigator)
>     in BottomTabNavigator (created by BottomTabs)
>     in BottomTabs (created by SceneView)
>     in StaticContainer
>     in EnsureSingleNavigator (created by SceneView)
>     in SceneView (created by RouteView)
>     in RCTView (created by View)
>     in View (created by DebugContainer)
>     in DebugContainer (created by MaybeNestedStack)
>     in MaybeNestedStack (created by RouteView)
>     in RNSScreen (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by InnerScreen)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by InnerScreen)
>     in InnerScreen (created by RouteView)
>     in RouteView (created by NativeStackViewInner)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by ScreenStack)
>     in RNSScreenStack (created by ScreenStack)
>     in ScreenStack (created by NativeStackViewInner)
>     in NativeStackViewInner (created by NativeStackView)
>     in RCTView (created by View)
>     in View (created by SafeAreaInsetsContext)
>     in SafeAreaProviderCompat (created by NativeStackView)
>     in NativeStackView (created by NativeStackNavigator)
>     in NativeStackNavigator (created by InsideStack)
>     in InsideStack (created by SceneView)
>     in StaticContainer
>     in EnsureSingleNavigator (created by SceneView)
>     in SceneView (created by CardContainer)
>     in RCTView (created by View)
>     in View (created by CardContainer)
>     in RCTView (created by View)
>     in View (created by CardContainer)
>     in RCTView (created by View)
>     in View
>     in CardSheet (created by Card)
>     in RCTView (created by View)
>     in View (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by PanGestureHandler)
>     in PanGestureHandler (created by PanGestureHandler)
>     in PanGestureHandler (created by Card)
>     in RCTView (created by View)
>     in View (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by Card)
>     in RCTView (created by View)
>     in View (created by Card)
>     in Card (created by CardContainer)
>     in CardContainer (created by CardStack)
>     in RNSScreen (created by AnimatedComponent)
>     in AnimatedComponent
>     in AnimatedComponentWrapper (created by InnerScreen)
>     in Suspender (created by Freeze)
>     in Suspense (created by Freeze)
>     in Freeze (created by DelayedFreeze)
>     in DelayedFreeze (created by InnerScreen)
>     in InnerScreen (created by Screen)
>     in Screen (created by MaybeScreen)
>     in MaybeScreen (created by CardStack)
>  ERROR  Warning: Internal React error: Attempted to capture a commit phase error inside a detached tree. This indicates a bug in React. Likely causes include deleting the same fiber more than once, committing an already-finished tree, or an inconsistent return pointer.
> ```

any update?? I get the same error when I updated my RN to .0.71.1

**Comment by KrisLau** [comment]
@Degan90 It's a problem with your code as indicated above:

- https://github.com/facebook/react-native/issues/32673#issuecomment-1280884078
- https://github.com/facebook/react-native/issues/32673#issuecomment-1280587000

**Comment by colaquecez** [comment]
I've fixed that by changing my listeners, with the new React Native version we need to change them.

![Screenshot 2023-02-05 at 4 24 00 AM](https://user-images.githubusercontent.com/38195454/216806918-b9fcfcb7-3c9d-437a-a688-25f0a7ed3d24.png)

**Comment by jarapamikaella** [comment]
any updates?

**Comment by trehman82** [comment]
any solution???????

**Comment by KrisLau** [comment]
@jarapamikaella @trehman82 Please read the thread: https://github.com/facebook/react-native/issues/32673?notification_referrer_id=NT_kwDOAsIbh7MyNzUwOTI2MDc0OjQ2Mjc1NDYz#issuecomment-1406970645.

**Comment by trehman82** [comment]
> @jarapamikaella @trehman82 and any future commenters. Please read the comments already left in this issue: #32673.
> 
> There are no solutions/updates posted because as far as I can tell this is a problem the way you are using `useEffect` so you need to debug your own code.

Thanks for reply. Actually I am not using useEffect in my project. All the components are class based. I don't know whats the matter.
Regards!

**Comment by zahid502** [comment]
> I've fixed that by changing my listeners, with the new React Native version we need to change them.
> 
> ![Screenshot 2023-02-05 at 4 24 00 AM](https://user-images.githubusercontent.com/38195454/216806918-b9fcfcb7-3c9d-437a-a688-25f0a7ed3d24.png)

I'm also facing the same issue.
I have tried your method but there there is the same issue, it's not solved.
@KrisLau @trehman82
Any update regarding to this issue?

**Comment by KrisLau** [comment]
@zahid502 I've already stated my answer numerous times in the thread above. No one can help you since it is likely a problem with your code. Do not ping specific commenters unless you are specifically replying to them.

**Comment by UnderTheMoonspell** [comment]
We just upgraded React Native from 0.69.7 to 0.71.3 and we started getting this in tons of Testing Library tests. Started by removing all the useEffects in a component to test a simple render test, and was still getting the error. So I stripped the component of anything besides a View inside a SafeAreaView and still getting the error in the test.

So now the Tree is simply this:

`  const testingLibraryAPI = render(
    <NavigationContainer>
        <Navigator>
          <NavigatorScreen
            name={'TESTScreen'}
            component={Splash}
          />
        </Navigator>
    </NavigationContainer>
  );`

Theres no error in the tests if I render the Component without the Navigator stack

**Comment by henaharon** [comment]
> The return in a `useEffect` should be a function: https://reactjs.org/docs/hooks-effect.html#:~:text=Effects%20with%20Cleanup. If you're getting an error, check what you're returning in the `useEffect`

Thank you so much!
migrating from RN 0.64 -> 0.71 and a lot of different things discovered.

**Comment by markobalogh** [comment]
@UnderTheMoonspell I'm also getting this error without anything weird happening in my `useEffect` calls, and it's happening when I navigate to a route in @react-navigation. So I think you're on to something...

**Comment by 001mbilal** [comment]
I thought the problem is that you guys, when calling event Listener ..
i.e-->> 
useEffect (e => { e.addEventListener}
 //you should do this..
const any_name = e => { e.addEventListener}

// and for clearing the event you should this!
return ()=> {
any_name.remove();
},[]
)

**Comment by 001mbilal** [comment]
The anothet possibilty is that when using addListener in ReactNative. i.e navigation.addListener() must need a route that you as a 2nd argument when you don't pass and destructure any route then that will be cause for undefined!

**Comment by jhsonmac** [comment]
> I've fixed that by changing my listeners, with the new React Native version we need to change them.
> 
> ![Screenshot 2023-02-05 at 4 24 00 AM](https://user-images.githubusercontent.com/38195454/216806918-b9fcfcb7-3c9d-437a-a688-25f0a7ed3d24.png)

I had already updated the listeners to:
⚠️⚠️⚠️⚠️⚠️⚠️
```
        Keyboard.addListener('keyboardDidShow', keyboardDidShow);
        Keyboard.addListener('keyboardDidHide', keyboardDidHide);

        // cleanup function
        return() => {
          Keyboard.removeAllListeners('keyboardDidShow');
          Keyboard.removeAllListeners('keyboardDidHide');
        };
```

But following @colaquecez suggestion worked great, problem solved.
🚀😁🎉😉👏🎊

```
        const unKeyboardDidShow = Keyboard.addListener('keyboardDidShow', keyboardDidShow);
        const unKeyboardDidHide = Keyboard.addListener('keyboardDidHide', keyboardDidHide);

        // cleanup function
        return() => {
          unKeyboardDidShow.remove();
          unKeyboardDidHide.remove();
        };
```

Thanks!

**Comment by KeaganStevens** [comment]
Upgrading from react-navigation 5 to react-navigation 6 solved the problem for me.

**Comment by Gilmara-Git** [comment]
@colaquecez suggestion worked for me as well.

```
const updateDeviceDim =  Dimensions.addEventListener("change", updateWidth);
    return ()=>{
      updateDeviceDim.remove(); 
    }

    // Dimensions.addEventListener("change", updateWidth);
    // return () => {
    //   Dimensions.removeEventListener("change", updateWidth);
    // };

Thank you :)
```

**Comment by alwex** [comment]
for those having an issue with react-navigation, I had the same issue after upgrading to RN 0.71.8, my app is using react-navigation and I implemented deep linking following their documentation for v5 here: https://reactnavigation.org/docs/5.x/deep-linking

The doc provide an example with this piece of code:
```ts
    return () => {
      // Clean up the event listeners
      Linking.removeEventListener('url', onReceiveURL); <=== HERE
      branch.unsubscribe();
    };
```
wich was causing the issue on my app.
Note that the latest version of the doc: https://reactnavigation.org/docs/deep-linking/ provide an updated example
```ts
    return () => {
      // Clean up the event listeners
      unsubscribeFirebase();
      linkingSubscription.remove(); <=== HERE
    };
```
Hope this can help

**Comment by jikseyres16** [comment]
> I've fixed that by changing my listeners, with the new React Native version we need to change them.
> 
> ![Screenshot 2023-02-05 at 4 24 00 AM](https://user-images.githubusercontent.com/38195454/216806918-b9fcfcb7-3c9d-437a-a688-25f0a7ed3d24.png)

You saved me, Thanks!

**Comment by sevmezabdullah** [comment]
`<ApiProvider api={userApi} setupListeners={false}>{children}</ApiProvider>`
for react native v0.72.3
I fixed by adding setupListeners={false} 
it worked for me.

**Comment by FloatingAnoj** [comment]
> I've fixed that by changing my listeners, with the new React Native version we need to change them.
> 
> ![Screenshot 2023-02-05 at 4 24 00 AM](https://user-images.githubusercontent.com/38195454/216806918-b9fcfcb7-3c9d-437a-a688-25f0a7ed3d24.png)

This fixed it for me. Good looking out. Thanks.

**Comment by Azam-Ra** [comment]
I have encountered the same error; furthermore, I've had an additional error:
`TypeError: undefined is not a function`, providing location of the error: `../node_modules/react-native-modal/dist/modal.js`. 
The invalid code was:
![image](https://github.com/facebook/react-native/assets/60223738/9ff63d38-0297-479d-88cb-0dcf2de8e3dd)
Instead of (correct code):
![image](https://github.com/facebook/react-native/assets/60223738/6ceba9f7-1644-4ba0-b450-6cf8309d57f9)

Turns out there is no such property as '`removeListeners`' on type '`DeviceEventEmitterStatic`'. Instead, the correct, existing property was `removeAllListeners`. However, note that editing code of the outsource library in node_modules is usually not a good idea.

I know it might seem as a stupid error, let me tell you, it is reaally stupid. But since I wasn't working on the project from the start, and also I was upgrading from RN version 0.66.5 to 0.72.3, I think it's acceptable. I just hope this will help someone to find a clue for the solution of their problem.
